### PR TITLE
Filter out more data provider traits and fix a rust link

### PR DESCRIPTION
**Pull request by sffc**
Some initial cleanup for data adapter FFI

### CODE REVIEW

File: ffi/diplomat/ffi_coverage/src/main.rs [hunk 1]
```diff
@@ -275,6 +289,9 @@ lazy_static::lazy_static! {
         // https://github.com/rust-diplomat/diplomat/pull/244
         "icu::locid::Locale::UND",
 
+        // Need to think about how to expose DataErrorKind for this to work
+        "icu_provider_adapters::empty::EmptyDataProvider::new_with_error_kind#FnInStruct",
```
Comment (sffc) [inline ffi/diplomat/ffi_coverage/src/main.rs:293]: @Manishearth  Why does this still show up in missing_apis.txt?

Comment (Manishearth) [reply ffi/diplomat/ffi_coverage/src/main.rs:293]: The pound sign is unnecessary

Comment (Manishearth) [inline ffi/diplomat/ffi_coverage/src/main.rs:293]:
```suggestion
        "icu_provider_adapters::empty::EmptyDataProvider::new_with_error_kind",
```

**Review by Manishearth — DISMISSED** [review]
r=me with that fixed

**Review by Manishearth — APPROVED** [review]

**Pull request merged**